### ISSUE TITLE
Values of the new (replaced) resource on "Resource view" are not displayed

### ISSUE THREAD

**Issue by JasminaPN** [ckan/ckan#4004]
#Tested on CKAN 2.6.4 and 2.7.2

Steps to reproduce the issue:
1. go to the following portal https://demo.ckan.org/
2. go to "Datasets" page and click "+Add Dataset Button"
3. Fulfill the fields for creating the Metadata of the dataset
4. At the bottom of the page click "Next: Add Data" button
5. On "Add Data" page add "Data" (resource ) by uploading a file (ex: CSV file)
6. Fulfill the other fields (optional)
7. Click "Finish" button - The dataset - datapusher-test was created - https://demo.ckan.org/dataset/datapusher-test
8. After the creation of the dataset, on the dataset (datapusher - test ) page click the link of the resource - Resource page will be opened.
9. On Resource page click "Manage" button - edit resource page will be opened
10. Click "remove" button on "File" field and upload new file: (another CSV file)
11. Click "Update Resource" button - new resource was replaced with the previous one.
12. After adding new resource on "resource view", old (from the previous - replaced) values are displayed instead of the values of the new uploaded resource. This can be fix on Datastore tab, on "manage' resource page by clicking "Upload to DataStore" button. 

Expected behavior: The values from the new resource should be displayed on "Resource View" page instead of the values of the previous (replaced) resource

**Comment by Engerrs** [comment]
Hi @JasminaPN .
I've tested the issue you described both on CKAN 2.6 and 2.7 (last versions), using installation both on localhost and on apache2, both of those variants are working well.
Some time ago on CKAN 2.6 I faced with a problem when client said the CSV don't chnage their values after updating CSV with new values, but it figures out that it take some time to Datapusher to push all the data in Datastore, especially for file with big amount of data, that's why I tried to check same behaviour on https://demo.ckan.org/, but have same problem that you've described in the issue.

My suggestion is that it can a problem with setup Datapusher on https://demo.ckan.org/, as other sites don't face such problem.
Nevertheless , I guess you can try to reproduce same problem locally first.

**Comment by tino097** [comment]
@Engerrs i can confirm that Datapusher needs some time to push the data and thus we are still seeing old data in the preview. 

However i've tested little bit different scenario:
 

```
1. User uploads resource A to the dataset.
2. Datapusher is notified and push the data from resource A to datastore
3. User replaces resource A with new resource B
4. Datapusher is notified and push the data from resource B to datastore
5. User replaces resource B with resource A
6. Datapuser is not notified and we still see data from resource B
```


If you can test this, please respond with your results

Thanks

**Comment by Engerrs** [comment]
Hi @tino097 , I've checked locally proposed scenario and it passed successfully.
Resource data is updated, but I could reproduce the actual issue you've described:
1. Create res A with data A.
2. Update res A with data B (this data should be big enough, I used CSV 4 mb with 36 635 rows in it).
3. After preview is changes and Datapusher didn't upload yet all data in Datastore, update the res A again with data A. Then the preview won't change and still show data B.
It's, because Datapusher didnt' upload all data yet to Datastore, that's why new uploading process didn't start.
